src/Entities/Entity.cpp

```diff
@@ -457,10 +457,15 @@ bool cEntity::DoTakeDamage(TakeDamageInfo & a_TDI)
 					case mtSilverfish:
 					{
 						a_TDI.RawDamage += static_cast<int>(ceil(2.5 * BaneOfArthropodsLevel));
-						// TODO: Add slowness effect
+						// The duration of the effect is a random value between 1 and 1.5 seconds at level I,
+						// increasing the max duration by 0.5 seconds each level, up to 3.5s with Bane of Arthropods V.
+						// Ref: https://minecraft.gamepedia.com/Enchanting#Bane_of_Arthropods
+						double MaxDuration = std::min(1 + BaneOfArthropodsLevel * 0.5, 3.5);
+						double Duration = GetRandomProvider().RandReal<double>(1, MaxDuration * 20);
+						Monster->AddEntityEffect(cEntityEffect::effSlowness, static_cast<int>(ceil(Duration)), 4);
```

Comment:
If I'm not mistaken this is a duration between 1 _tick_ and `MaxDuration` seconds so that should be something like `RandReal<double>(1, MaxDuration) * 20`.
Although I think it would be better to generate an `int` instead of using `ceil(double)` which I think biases against the minimum value.